### PR TITLE
refactor: simplify ctx API

### PR DESCRIPTION
**Pull request by antongolub**
- [x] Tests pass
- [x] Types updated

closes #416

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -148,4 +149,7 @@ declare module 'zx/experimental' {
 
   type StopSpinner = () => void
   export function startSpinner(title: string): StopSpinner
+
+  export const getCtx: AsyncLocalStorage<any>['getStore']
```
Comment (antonmedv) [inline src/index.d.ts:153]: This is pretty basic. any is not any types at all.

Comment (antonmedv) [reply src/index.d.ts:153]: Let’s wait for my idea.

**Comment by antongolub** [comment]
> Let’s wait for my idea. 

I just want to see where my own thinking takes me )

File: src/index.d.ts (hunk 1 above)
Comment (antongolub) [reply src/index.d.ts]: I just want to see where my own thinking takes me )

Comment (antongolub) [reply src/index.d.ts]:
```js
$.block(($) => {
   $.verbose = false
   fetch('https://example.com')
})````

**Comment by antongolub** [comment]
It works!

```js
  $.o = (opts) => (...args) => ctx(($) => Object.assign($, opts)(...args))

  const quiet = (...args) => {
    if (args[0] instanceof ProcessPromise) {
      args[0].ctx.verbose = false
      return args[0]
    }
    return $.o({verbose: false})(...args)
  }

  await quiet`echo foo`
  await quiet($`echo foo`)
```

**Comment by antongolub** [comment]
@antonmedv,

`ctx()` rocks! #385 chain API proposal is here:

```js
  try {
    await quiet(timeout(100, 'SIGKILL')`sleep 9999`)
  } catch {
    console.log('killed1')
  }

  try {
    const p = $`sleep 9999`
    await quiet(timeout(100, 'SIGKILL')(p))
  } catch {
    console.log('killed2')
  }

  try {
    await $`sleep 9999`.quiet().timeout(100, 'SIGKILL')
  } catch {
    console.log('killed3')
  }
```

**Comment by antongolub** [comment]
Let's introduce labels: v6, experimental, docs, etc

**Review by antonmedv — APPROVED** [review]

**Pull request merged**